lib/logstash/outputs/elasticsearch.rb

```diff
@@ -387,7 +387,7 @@ def handle_event_mapping_errors(event_mapping_errors)
 
     event_mapping_errors.each do |event_mapping_error|
       detailed_message = "#{event_mapping_error.message}; event: `#{event_mapping_error.event.to_hash_with_metadata}`"
-      handle_dlq_status(event_mapping_error.event, :warn, detailed_message)
+      @dlq_writer ? @dlq_writer.write(event_mapping_error.event, detailed_message) : @logger.warn(detailed_message)
```

Comment:
In case of json log format, it logs a json document with `message` containing the event map converted to a string.
Is that desired or do you think it should keep document structure, nested in a `event` field, like you suggested in [comment](https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1104#discussion_r1054681500)?

I like to keep things simple, without many moving parts, but this reintroduces a "code duplication"
```ruby
if @dlq_writer
  @dlq_writer.write
else 
    @logger.log
end    
```  

that the original #1084 tried to avoid.

Reply:
The original logging for _dropped-events-when-DLQ-was-unavailable_ included the action tuple that was the cause of a rejection from Elasticsearch, but these events are being routed to DLQ because they _cannot_ generate a valid action tuple -- we only have our own pre-request rejection reason and the event itself.

While #1084 tried to avoid this particular DLQ-or-LOG duplication, the related #1080 _also_ introduced paths in this method that differ based on the availability of DLQ -- the abstraction has leaked already.

Looking at this again, it also doesn't make much sense for our `detailed_message` that is being added to the DLQ as _metadata-about-why-the-event-was-invalid_ to contain a stringified version of the entire event, since the entry itself will contain the entire event.

What about this (pseudo-code):

~~~ ruby
error_message, event = event_mapping_error.message, event_mapping_error.event
@dlq_writer ? @dlq_writer.write(event, error_message)
            : @logger.warn("Could not index event to Elasticsearch: #{error_message}", event: event.to_hash_with_metadata)
~~~

In this way, the DLQ entry would have a message like:
> ~~~
> "Badly formatted index, after interpolation still contains placeholder: [%{indexprefix}-foo]"
> ~~~
> ~~~
> "Elasticsearch doesn't support [DANCE] action"
> ~~~

And when the DLQ is disabled the logger would have entries like:
> ~~~
> {"message":"Could not index event to Elasticsearch: Badly formatted index, after interpolation still contains placeholder: [%{indexprefix}-foo]", "event": { ... } }
> ~~~
> ~~~
> {"message":"Could not index event to Elasticsearch: Elasticsearch doesn't support [DANCE] action", "event": {...} }
> ~~~

Reply:
I like it, go for it :+1: